.buildkite/metricbeat/pipeline.yml

```
@@ -1,5 +1,153 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+env:
+  IMAGE_UBUNTU_X86_64: "family/core-ubuntu-2204"
+  IMAGE_UBUNTU_ARM_64: "core-ubuntu-2004-aarch64"
+  IMAGE_WIN_2016: "family/core-windows-2016"
+  IMAGE_WIN_2019: "family/core-windows-2019"
+  IMAGE_WIN_2022: "family/core-windows-2022"
+  IMAGE_MACOS_X86_64: "generic-13-ventura-x64"
+  PIPELINE_NAME: "metricbeat"
 
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+
+  - input: "Input Parameters"
+    key: "input-run-all-stages"
+    fields:
+    - select: "Metricbeat - runAllStages"
+      key: "runAllStages"
+      options:
+        - label: "True"
+          value: "true"
+        - label: "False"
+          value: "false"
+      default: "false"
+    - select: "Metricbeat - runMacOsTests"
+      key: "UI_MACOS_TESTS"
+      options:
+        - label: "True"
+          value: "true"
+        - label: "False"
+          value: "false"
+      default: "false"
+    if: "build.source == 'ui'"
+
+  - wait: ~
+    if: "build.source == 'ui'"
+    allow_dependency_failure: false
+
+  - group: "Mandatory Tests"
+    key: "mandatory-tests"
+    if: ${ALLOW_MANDATORY_TESTS}
+    steps:
+      - label: ":linux: Ubuntu Unit Tests"
+        key: "mandatory-linux-unit-test"
+        command: ".buildkite/scripts/unit_tests.sh ${PIPELINE_NAME}"
```

I wonder whether this command could directly use the `mage build unitTest` and install the required tools using `.buildkite/hooks/pre-command` by calling `source .buildkite/scripts/install_tools.sh`?

By doing so, all the specific files in `.buildkite/scripts/` for running the mage commands won't be needed.

A similar question for the other steps.

---- 

To help with my proposal:

```suggestion
        command: "cd ${PIPELINE_NAME} && mage build unitTest"
```

**NOTE**: unfortunately, the [Buildkite command](https://buildkite.com/docs/pipelines/command-step) does not allow to set where to run the given command from natively, but with a [plugin](https://github.com/thedyrt/change-directory-buildkite-plugin) - that's something GitHub actions support by default with the [working-directory](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsworking-directory)

---

changed the logic a bit, please take a look at that again

---

What do you think something like the below snippet?

```diff
diff --git a/.buildkite/hooks/pre-command b/.buildkite/hooks/pre-command
index b8c2620260..e690e5dea3 100644
--- a/.buildkite/hooks/pre-command
+++ b/.buildkite/hooks/pre-command
@@ -53,3 +53,13 @@ if [[ "${ONLY_DOCS}" == "false" && "${BUILDKITE_PULL_REQUEST}" != "" ]] || [[ "$
     fi
   fi
 fi
+
+
+# Configure the requierd tools for running the metricbeat pipeline
+if [ "$BUILDKITE_PIPELINE_SLUG" == "beats-metricbeat" ]; then
+  source .buildkite/scripts/install_tools.sh
+
+  # NOTE: What's the rationale for doing this?
+  sudo chmod -R go-w "${BEATS_PROJECT_NAME}/"
+  umask 0022
+fi
\ No newline at end of file
diff --git a/.buildkite/metricbeat/pipeline.yml b/.buildkite/metricbeat/pipeline.yml
index 0a9383f448..f592e7e057 100644
--- a/.buildkite/metricbeat/pipeline.yml
+++ b/.buildkite/metricbeat/pipeline.yml
@@ -40,12 +40,12 @@ steps:
     steps:
       - label: ":linux: Ubuntu Unit Tests"
         key: "mandatory-linux-unit-test"
-        command: ".buildkite/scripts/unit_tests.sh"
+        command: "cd ${BEATS_PROJECT_NAME} && mage build unitTest"
         agents:
           provider: "gcp"
           image: "${IMAGE_UBUNTU_X86_64}"
           machineType: "c2-standard-16"
-        artifact_paths: "metricbeat/build/*.*"
+        artifact_paths: "$BEATS_PROJECT_NAME/build/*.*"
 
       - label: ":go: Go Intergration Tests"
         key: "mandatory-int-test"
diff --git a/.buildkite/scripts/unit_tests.sh b/.buildkite/scripts/unit_tests.sh
deleted file mode 100755
index 95ea995106..0000000000
--- a/.buildkite/scripts/unit_tests.sh
+++ /dev/null
@@ -1,14 +0,0 @@
-#!/bin/bash
-
-source .buildkite/scripts/install_tools.sh
-
-set -euo pipefail
-
-echo "--- Run Unit Tests"
-sudo chmod -R go-w "${BEATS_PROJECT_NAME}/"
-pushd "${BEATS_PROJECT_NAME}" > /dev/null
-
-umask 0022
-mage build unitTest
-
-popd > /dev/null
```

I deleted `.buildkite/scripts/unit_tests.sh` and called `source .buildkite/scripts/install_tools.sh` in the `.buildkite/hooks/pre-command` so I could then change the `.buildkite/metricbeat/pipeline.yml` to use `command: "cd ${BEATS_PROJECT_NAME} && mage build unitTest"`

---

IIUC, all the scripts are sharing the same content but with the difference in the `mage` target. Then, the `pre-command` hook can be the one centralising all that logic and the pipeline calling directly the `mage` command in the given directory/beat

---

Unfortunately, the `pre-command` hook doesn't work as we need, for example, the command `sudo` returns the error, and so on. It could be the source of other issues. It will be solved by using our own images for BK. But before that, we have to finish with the migration of all the small pipelines and as a result, we will have a list of tools for our own image(s).

---

> I suggest don't block the migration if all steps of the pipeline work properly.

I won't block anything; I'm just helping with the review. 👼 


I'll add a few more comments, but overall, I think I already pointed out the following (in my previous comments):
- centralise configuration in the `pre-command` as much as possible - BK images might simplify those configurations.
- easy conditions about what/how to trigger a build 
  - so far, I see too many nested conditions for creating those conditions.
  - The pipeline could run the mage/make command directly rather than using intermediate scripts.